inject.js

```diff
@@ -277,19 +277,36 @@ chrome.runtime.sendMessage({}, function(response) {
         }, true);
       });
 
+      var timer_new = null, last_vscid = null;
+
+      function deleteCtrl(node) {
+        let id = node.dataset['vscid'];
+        let ctrl = document.querySelector(`div[data-vscid="${id}"]`)
+        if (ctrl) {
+          ctrl.remove();
+        }
+        node.classList.remove('vsc-initialized');
+        delete node.dataset['vscid'];
+      }
+
       function checkForVideo(node, parent, added) {
         if (node.nodeName === 'VIDEO') {
           if (added) {
-            new tc.videoController(node, parent);
+            if (timer_new) {
+              clearTimeout(timer_new);
+              timer_new = null;
+            } else {
+              new tc.videoController(node, parent);
+            }
           } else {
             if (node.classList.contains('vsc-initialized')) {
-              let id = node.dataset['vscid'];
-              let ctrl = document.querySelector(`div[data-vscid="${id}"]`)
-              if (ctrl) {
-                ctrl.remove();
+              if (timer_new) {
+                // already waiting other one, delete this one immediately
+                deleteCtrl(node)
+              } else {
+                // wait for new one
+                timer_new = setTimeout(deleteCtrl, 10, node)
```

Comment:
Hmm, not sure I follow how this works.. There is no guarantee that video element being removed is the same element that's being added later to the DOM, so how does setting the timeout here help address the problem? 

I feel like I'm missing something obvious :)